### PR TITLE
control_demo_robot_headless package + initial structure

### PR DESCRIPTION
**Pull request by olivier-stasse**
This PR is a request for comments.Based on the calls it tries to rewrite the ros_control_boiler_plate repository using ros2 and ros2_control concept. It implements:
  * Initialize the CMake infrastructure / metapackage
  * Add the package control_demo_robot_headless:
      * Creates GenericRobotHardware class.
      * Add reading ```/robot_publisher_state/robot_description``` parameter using a synchronous method.
      * Add callback method to read ```/robot_description``` topic
  * Add a gtest test_generic_robot_hardware
  * Add a .clang_format file to make the cpplinters happy.

### CODE REVIEW

File: ros2_control_demo_robot_headless/include/control_demos_robot_headless/generic_robot_hardware.hpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  /* \brief Perception: Read sensors.
+   */
+  hardware_interface::hardware_interface_ret_t read();
+  /* \brief Action: write the command to the low-level hardware
+   */
+  hardware_interface::hardware_interface_ret_t write();
+
+ protected:
+  /* \brief The robot model in URDF format.
+     Should we keep the string if more informations are store in the URDF ?
```
Comment (bmagyar) [inline ros2_control_demo_robot_headless/include/control_demos_robot_headless/generic_robot_hardware.hpp:54]: I think that for now we can keep it, let's adjust the text
```suggestion
    Keep the string to allow additional parsing of URDF
```

**Comment by olivier-stasse** [comment]
I'll let @destogl take the lead on this.

**Comment by destogl** [comment]
@olivier-stasse Thank you for your contribution. This really helped me a lot to crate #4 . As I explained yesterday it was not my intention to "reject" your contribution. The reason for the #4 is to clarify my thoughts and make the structure with CI, linters, code coverage enabled before we add any code. I cherry-picked commits from your branch for this with the basics. This is a very nice input and enabled me to create the structure faster.

@bmagyar Thank you for starting the review. I should have communicated my intention earlier, but I want first to have something "real" and test it before start to confuse people I wasn't sure this will work as intended. Nevertheless, I will take care to communicate things in the future as early as possible.

File: ros2_control_demo_robot_headless/package.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>ros2_control_demo_robot_headless</name>
+  <version>0.0.0</version>
+  <description>ROS2 control demo robot headless</description>
```
[the remark this answers is not in the document]
Comment (destogl) [reply ros2_control_demo_robot_headless/package.xml:5]: I will adapt this to #4

File: ros2_control_demos/package.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0"?>
+<package format="3">
+  <name>ros2_control_demos</name>
+  <version>0.0.0</version>
+  <description>Metapackage for ROS2 control demos packages</description>
+  <maintainer email="allabana@amazon.com">Anas Abou Allaban</maintainer>
```
[the remark this answers is not in the document]
Comment (destogl) [reply ros2_control_demos/package.xml:6]: Will also be removed from #4

**Pull request closed without merging**